### PR TITLE
Proposal and docs for changes to snapshot layout

### PR DESCRIPTION
**Pull request by kensimon**
This functions as both a proposal for how the snapshot tarballs should
be structured, as well as documentation for that structure.

Places with a "NOTE:" section are places where changes are proposed.

### CODE REVIEW

**Comment by abiogenesis-now** [comment]
@kensimon I have some screenshots that I think makes this info easier to digest, e.g. see below for `/plugins` structure:

screen shot 2017-08-08 at 4 41 48 pm

Once the structure is nailed down, let me know if you think these are helpful to incorporate. I can push them onto this branch if that's easiest.

^also, will we be unzipping nested tarballs?

**Comment by kensimon** [comment]
Sure, screenshots would be awesome.  Up to you if you want to put them before or after we merge... I think so long as it's clear the document is a work-in-progress we can merge early and often on this one.

The tar-within-tar problem will be fixed in a separate PR, I think we can update the documentation then, although I don't think the docs will have to change.  Reason being: I think the fix for plugins that send tarfiles will be to implicitly extract them before putting them in the main tarball... that way, for someone consuming a sonobuoy snapshot, they'll never see a nested tarball in the first place (thus it's probably fine that the docs don't mention them.)

**Comment by abiogenesis-now** [comment]
Makes sense! (re the tar-within-tar)
Screenshots added, feel free to edit.

**Comment by kensimon** [comment]
@timothysc if you're happy with this, feel free to approve (or just merge), I think we can iterate more if there's feedback. I have another PR that implements these changes too, but I'll wait for us to have a consensus on this first.

**Comment by timothysc** [comment]
@kensimon Let give this some time for other folks and I'll send out an email too.

File: docs/snapshot.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+![tarball hosts screenshot][5]
+
+### /podlogs
+
+> NOTE: pod logs are currently distributed throughout the tarball in `/resources/ns/<namespace>/pods/<podname>/<containername>.txt`.  The proposal is to move them under a different directory so that "types" within a directory are consistent
+
+The `/podlogs` directory contains logs for each pod found during the Sonobuoy run, similarly to what you would get with `kubectl logs -n <namespace> <pod> <container>`.
+
+- `/podlogs/<namespace>/<podname>/<containername>.log` - Contains the logs for the each container, for each pod in each namespace.
```
Comment (timothysc) [inline docs/snapshot.md]: In your example image I don't see podlogs namespaced.

File: docs/snapshot.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+The top-level directories in the results tarball look like the following:
+
+![tarball overview screenshot][3]
+
+### /resources
+
+The `/resources` directory contains JSON-serialized Kubernetes objects, taken from querying the Kubernetes REST API. The directory has the following structure:
+
+- `/resources/ns/<namespace>/<type>.json` - For all resources that belong to a namespace, where `<namespace>` is the namespace of that resource (eg. `kube-system`), and `<type>` is the type of resource, pluralized (eg. `Pods`).
+- `/resources/cluster/<type>.json` - For all resources that don't belong to a namespace, where `<type>` is the type of resource, pluralized (eg. `Nodes`).
```
Comment (timothysc) [inline docs/snapshot.md]: Your image shows non-ns vs. cluster

Comment (abiogenesis-now) [inline docs/snapshot.md]: I'll fix that

Comment (abiogenesis-now) [inline docs/snapshot.md]: done

File: docs/snapshot.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+This looks like the following:
+
+![tarball plugins screenshot][7]
+
+### /meta
+
+The `/meta` directory contains metadata about this Sonobuoy run, including configuration and query runtime.
+
+- `/meta/query-time.json` - Contains metadata about how long each query took, example: `{"queryobj":"Pods","time":12.345ms"}`
+  > NOTE: this file is currently distributed throughout the tarball in `/resources/.../results.json`.  Proposal is to move it here, and create other `.json` files under `/meta` if we start capturing other things than query time.
```
Comment (timothysc) [inline docs/snapshot.md]: We'll have to see if this is problematic or not.

File: docs/snapshot.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+- `/meta/config.json` - A copy of the Sonobuoy configuration that was set up when this run was created, but with unspecified values filled in with explicit defaults, and with a `UUID` field in the root JSON, set to a randomly generated UUID created for that Sonobuoy run.
+
+This looks like the following:
+
+![tarball meta screenshot][8]
+
+### /serverversion.json
+
+> NOTE: this is currently `/serverversion/serverversion.json`, proposal here is to just make it `/serverversion.json`
```
Comment (timothysc) [inline docs/snapshot.md]: Why not cluster ?

**Comment by timothysc** [comment]
@kensimon or @abiogenesis-now feel free to rebase then merge.

**Comment by abiogenesis-now** [comment]
FYI added the snapshot file to the docs table-of-contents

**Pull request merged**